docs/developer-docs/security/security-best-practices/inter-canister-calls.mdx

```diff
@@ -269,11 +279,71 @@ mod test {
 }
 ```
 
+</TabItem>
+
+<TabItem value="motoko" label="Motoko">
+
+```motoko
+import Result "mo:base/Result";
+import Map "mo:base/HashMap";
+import Error "mo:base/Error";
+import Principal "mo:base/Principal";
+
+actor {
+
+  let pending_requests = Map.HashMap<Principal, Bool>(10, Principal.equal, Principal.hash);
+
+  private func guard(principal : Principal) : Result.Result<Principal, Error> {
+    if (pending_requests.get(principal) != null) {
+      #err(Error.reject("Already processing a request for principal " # Principal.toText(principal)));
+    } else {
+      pending_requests.put(principal, true);
+      #ok principal;
+    };
+  };
+
+  private func drop(principal : Principal) : () {
+    ignore pending_requests.remove(principal);
+  };
+
+  public shared ({caller}) func example_call_with_locking_per_caller() : async Result.Result<(), (Error.ErrorCode, Text)> {
+    var success = false;
+    try {
+      // Try to create a lock for `caller`, return an error immediately if there is already a call in progress for `caller`
+      switch (guard(caller)) {
+        case (#err e) {
+          return #err(Error.code(e), Error.message(e));
+        };
+        case (#ok _) {
+          success := true;
+        };
+      };
+      // do anything, call other canisters
+      #ok;
+    } catch e {
+      #err(Error.code(e), Error.message(e));
+    } finally {
+      // Release the guard (only request's callbacks that have created a block)
+      if success {
+        drop caller;
```

Comment:
nit: WDYT about changing the name to `drop_guard` to make even clearer what happens here?